app/src/main/java/in/ac/bits_pilani/goa/ard/activities/MainActivity.java

```diff
@@ -77,6 +125,41 @@ public void onClick(final View view) {
         toggle.syncState();
 
         navigationView.setNavigationItemSelectedListener(this);
+        navDrawerDBRef = FirebaseDatabase.getInstance().getReference(AHC.FDR_NAV_DRAWER);
+
+        final View headerView = navigationView.getHeaderView(0);
+        final ImageView navDrawerImage = ButterKnife.findById(headerView, R.id.nav_drawer_image);
+        final TextView navDrawerTitle = ButterKnife.findById(headerView, R.id.nav_drawer_title);
+        final TextView navDrawerSubtitle = ButterKnife.findById(headerView, R.id.nav_drawer_subtitle);
+
+        if (navDrawerTitleText != null) {
+            navDrawerTitle.setText(navDrawerTitleText);
```

Comment:
Yes, but since it is static, when onCreate is called next time (ex- when orientation changes), it will have gotten value from earlier instance.
A sample life cycle-
```
<onStart>
navDrawerTitleText null  : nothing happens
onDataChange call
    update titleTextView
    set navDrawerTitleText
<orientation changes>
<onDestroy>
<onStart>
navDrawerTitleText "SomePreviousValue"   -> update TitletextView
onDataChange call
    if titleText has changed in db:
         update titleText
         update navDrwaerTitleText
```
Doing this for titleText and subTitleText doesn't have any practical use
However, for the background image to be same in one life cycle (unless db changes), the corresponding task for `navDrawerImageURL` in mainActivity is necessiry.